consumer/consumererror/signalerrors.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// TraceError is an error that may carry associated Trace data for a subset of received data
+// that faiiled to be processed or sent.
+type TraceError struct {
+	error
+	failed pdata.Traces
+}
+
+// Traces creates a TraceError that can encapsulate received data that failed to be processed or sent.
+func Traces(err error, failed pdata.Traces) error {
```

Comment:
We have two options `Traces` the ctor and `TracesError` the struct, or `Traces` the struct and `NewTraces` the ctor.

I think the second option is more consistent with the rest of the collector code, what do you think?

Reply:
I'll defer to you as you are undoubtedly more familiar with the rest of the collector code. I went with the names I did as they fit what had been in that package, but the whole point of this effort is to clean up that package so a change is definitely on the table. `Traces` for the struct and `NewTraces` for the initializer would be in line with the general Go idiom as well.